CHANGELOG/v1.7.0.md

```diff
@@ -0,0 +1,433 @@
+- e2e: Use manager in test extension (#10097)
+- Machine: Be more explicit when skipping node deletion (#10137)
+- Machine: Mark Machine healthy condition as unknown if we can't list wl nodes (#9864)
+- MachineDeployment: Set revision on machinedeployment and patch via defer (#10160)
+- MachinePool: Use SSA Patch to create machines in MP controller (#9791)
+- MachineSet: improve replica defaulting for autoscaler (#9649)
+- MachineSet: KCP/MS remove hardcoded WithStepCounterIf(false) (#9772)
+- Release: Add integration test for release notes tool (#9617)
+- Release: Bump kubebuilder-release-tools to v0.4.3 (#9818)
+- Release: Changelog entry to include cert-manager to v1.14.2 (#10188)
+- Release: Determine release type from tag to also handle beta releases (#10324)
+- Release: Fix dockerfile for clusterctl build (#10058)
+- Release: Fix grammar in release script (#9981)
+- Release: Improve release-staging build (#10095)
+- Release: Improve weekly PR update generation script and documentation (#10092)
+- Release: Make release notes tool not dependent on local git (#9618)
+- Release: Prep main branch for v1.7 development (#9799)
+- Release: Provide faster support for new k8s minor releases (#9971)
+- Release: Read in dockerfiles from stdin (#9990)
+- Release: also detect alpha releases as pre releases (#10379)
+- Release: Release v1.5.7 (#10251)
+- Release: Release v1.6.3 (#10252)
+- Release: Swap in new 1.7 RT members in OWNERS (#9855)
+- Runtime SDK: Add more helper functions in topologymutation varaible.go to help unmarshal variables (#9670)
+- Runtime SDK: Enable integration tests of RuntimeExtensions (#10330)
+- Testing: Add scale testing for upgrades (#9077)
+- Testing: Drop policy v1beta1 API support in the testing framework (#10158)
+- Testing: Drop unused runtime features for the in-memory provider (#9778)
+- Testing: Fix typo in test framework (#9873)
+- Testing: Fixing kubetest warnings for deprecations (#10172)
+- Testing: Print conformance image used in kubetest (#10076)
+- Testing: Remove k8s.io/utils/pointer from repository (#9836)
+- Testing: Small improvements to the inmemory api server (#9935)
+- Testing: add dynamic finalizer assertions for e2e framework (#9970)
+- Testing: Watch for Cluster resources in topology MachineSet & MachineDeployment controllers (#10029)
+- Testing: Watch for Cluster resources in topology MD controller (#9865)
+- util: Improve patch helper error handling (#9946)
+- util: Use min/max funcs from Go SDK (#9945)
+
+:book: Additionally, there have been 53 contributions to our documentation and book. (#10005, #10031, #10040, #10061, #10066, #10068, #10084, #10099, #10100, #10115, #10122, #10170, #10174, #10194, #10239, #10257, #10268, #10288, #10289, #10323, #10329, #10334, #10381, #10393, #10410, #9585, #9640, #9767, #9771, #9779, #9782, #9786, #9794, #9797, #9801, #9817, #9829, #9831, #9838, #9856, #9866, #9867, #9868, #9876, #9896, #9897, #9908, #9941, #9949, #9957, #9961, #9972, #9993) 
+
+## Dependencies
```

Comment:
I wonder if we should mention that this is for our main go module (maybe a short sentence below?)

Reply:
I just added `for the main go module` to the sentence. Would that make sense or is more info needed in a separate sentence?

Reply:
That's fine

Reply:
Oh wait. I meant to the dependencies section not to the book (sorry selected the wrong lines)

Reply:
If we can't easily add it to the Dependencies section because of the way that is generated, it's fine